api/controllers/document.js

```diff
@@ -306,6 +306,8 @@ exports.publicDownload = function (args, res) {
         }
         var fileMeta;
 
+        // clean the filename
+        fileName = encodeURIComponent(fileName).replace(/\(/g, '%28').replace(/\)/g, '%29').replace(/\\/g, '_').replace(/\//g, '_').replace(/\%2F/g, '_').replace(/ /g, '_');
```

Comment:
It's good that it matches the encoding in the other applications. There is a better way to do this than all of the replaces. Making a ticket to update the encoding in all of the applications.